docs/DockerContainerLifecycle.md

```diff
@@ -0,0 +1,340 @@
+
+The profile's `tags` data is stored as:
+
+    # /calico/v1/policy/profile/<profile_name>/tags
+    ["PROFILE_NAME", "TAG_A", "TAG_B", ...]
+
+Tags apply to all endpoints associated with this profile, and tags can be used 
+in the rules (as shown above).
+
+Here is the output from `calicoctl profile PROF_A_B rule show` for the default 
+rules:
+
+    Inbound rules:
+       1 allow from tag PROF_A_B
+    Outbound rules:
+       1 allow
+
+This means that the endpoints associated with this profile will be able to send 
+outbound traffic to anywhere, but the containers can only be reached by other 
+containers associated with this profile.
+
+For more information on how to create, remove, update, and show profiles, see 
+the [`calicoctl profile` reference guide](calicoctl/profile.md).
+
+### Result
+After creating the profile, there is still no connectivity between the 
+containers because the profile has not yet been associated with any endpoints.
+
+![calicoctl profile add](diagrams/profile_add.png)
+
+## Update Containers to use the Profile
+
+Setting a profile on a container provides connectivity to the container 
+relative to the policy rules of the associated profile(s).  The default rules 
+for a profile provide connectivity between endpoints on the same profile.
+
+**NOTE**: Felix deals with endpoints rather than containers, but for simple
+containers with a single interface managed using calicoctl, we treat a 
+container and endpoint as the same thing.  For more complicated scenarios, 
+calicoctl provides commands for managing actual endpoints (see the 
+[`calicoctl endpoint` reference guide](calicoctl/endpoint.md) for usage and 
+examples).
+
+### User Configures the Profile on each Container
+
+The profile is set on each workload using `calicoctl`:
+
+    # Host1
+    calicoctl container workload_A profile set PROF_A_B
+    
+    # On Host2
+    calicoctl container workload_B profile set PROF_A_B
+
+### Calico Actions
+When this command is run, the calicoctl tool sets `PROF_A_B` in the endpoint's 
+list of profile ids in etcd.
+
+Felix picks up this change to the endpoint and uses the rules of the profile in 
+etcd to configure ACLs in the Linux ip tables.
+
+Since this profile is using default rules and has not been modified, the ACLs 
+that Felix programs allow all incoming traffic from containers using `PROF_A_B` 
+and drop all other inbound traffic.  The ACLs also allow all outbound traffic 
+from the containers to any destination.
+
+### Result
+Now that the ip tables rules have been configured to allow specific incoming 
+traffic, connectivity has been achieved between endpoints!  The containers are 
+now able to send any kind of traffic to one another.
+
+![calicoctl container set profile](diagrams/set_profile.png)
+
+## More complicated policy
+
+Calico can go far beyond configuring simple node-to-node connections with one 
+profile using basic rules.
```

Comment:
The clear follow-on document is "Advanced Network Policy", no? Rather than a generic follow-on paragraph below, I'd call out this document in particular, and *maybe* provide a link back to the README.